lib/srv/tcpip/helpers.go

```diff
@@ -0,0 +1,86 @@
+/*
+ * Teleport
+ * Copyright (C) 2023  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package tcpip
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+
+	"github.com/gravitational/trace"
+)
+
+const (
+	// maxLengthPrefixedMessageSize is the upper limit of length prefixed messages that we will permit
+	// encoding/decoding. Note that this value is actually much larger than required for the usecase that
+	// these helpers were written to support.
+	maxLengthPrefixedMessageSize = 1024 * 64
```

Comment:
Technically a `u16` can't encode `1024 * 64`, it can only encode up to `1024 * 64 - 1` lol.  But more seriously, I picked `u32` because it felt a bit more future-proof.  Plus, GRPC delimits messages via `u32` length-prefixing so it must be a good idea.